parkbenchcommon/ramdisk.py

```diff
@@ -0,0 +1,132 @@
+        """Mounts the ramdisk. Raises an exception on failure, and does nothing if the disk
+        is already mounted.
+
+        size: The size of the ramdisk to be mounted. This should be a string representing a
+          number of bytes, and may include the single-character suffixes k, m, g, or % for
+          kibibytes, mebibytes, gibibytes, or percentage of physical RAM, respectively.
+        uid: The system user ID that should own the mount directory.
+        gid: The system group ID that should be associated with the mount directory.
+        mode: The mode of the mount directory access permissions. This should be a decimal
+          integer. ex: 511, not 777.
+          Note: This integer is most easily obtained by ORing the appropriate permission
+          flags from the stat module.
+        """
+
+        # Since size should be an integer with a single character suffix, we validate only
+        #   the integer part by including all but the last character of the string.
+        self._validate_size_option(size)
+        self._validate_integer_option('uid', uid)
+        self._validate_integer_option('gid', gid)
+        self._validate_integer_option('mode', mode)
+
+        # TODO #16: On Python 3 migration, oct() will break this code.
+        mount_options = 'size=%s,uid=%s,gid=%s,mode=%s' % (size, uid, gid, oct(mode))
+
+        if not self.is_mounted():
+            if os.path.isfile(self.path):
+                raise RamdiskMountError(
+                    'Could not mount ramdisk on %s. Specified path is a file.' % self.path)
+
+            if not os.path.isdir(self.path):
+                raise RamdiskMountError(
+                    'Could not mount ramdisk on %s. Specified path does not exist.' % \
+                    self.path)
+
+            if not os.listdir(self.path) == []:
+                self.logger.warning('Ramdisk mountpoint %s is not empty.', self.path)
+
+            return_code = subprocess.call(
+                ['mount', '-t', 'tmpfs', '-o', mount_options, 'none', self.path])
+
+            if not self.is_mounted():
+                # TODO #16: Implement exception chaining when we move to Python 3.
+                raise RamdiskMountError(
+                    'Could not mount ramdisk on %s. Mount return code was %s.' %
+                    (self.path, return_code))
+
+    def is_mounted(self):
+        """Checks whether the ramdisk is mounted.
+
+        Returns True if is mounted, returns False otherwise.
+        """
+        return 'none on %s type tmpfs' % self.path in str(subprocess.check_output('mount'))
+
+    def _validate_integer_option(self, option_name, value):
+        """Raises an exception and writes a log message if the given value is not an
+        integer.
+        """
+        if not self._is_integer(value):
+            message = 'The value %s for ramdisk mount option %s was not formatted' \
+                ' correctly.' % (value, option_name)
```

Comment:
I've been putting the empty space at the end of the prior line. I guess we don't have to be consistent.